src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs

```diff
@@ -251,7 +260,27 @@ public void Pack(string nuspecPath, string nupkgPath, Manifest manifest, bool pa
 
                 string baseDirectoryPath = (string.IsNullOrEmpty(BaseDirectory)) ? Path.GetDirectoryName(nuspecPath) : BaseDirectory;
                 builder.Populate(manifest.Metadata);
-                builder.PopulateFiles(baseDirectoryPath, manifest.Files);
+                builder.PopulateFiles(baseDirectoryPath, manifest.Files.Where(m =>
+                {
+                    if (!IgnoreMissingFiles)
+                    {
+                        return true;
+                    }
+
+                    PackageBuilder temp = new PackageBuilder();
+                    bool didAdd = false;
+
+                    try {
+                        temp.AddFiles(baseDirectoryPath, m.Source, m.Target, m.Exclude);
```

Comment:
ManifestFile is a thin wrapper around the `<File>` entry in the Manifest.  That means that m.Source may not be a flie, but rather a wildcard pattern (including `**` style globs) so there's no good way to test if that exists, that I've found, short of letting NuGet do its resolution.